components/tabs/src/sync/bridge.rs

```diff
@@ -105,26 +97,23 @@ impl BridgedEngine for BridgedEngineImpl {
         Ok(sync_id.to_string()) // this is a bit odd, why the result?
```

Comment:
This isn't at all related to your change, but I saw this comment and have an answer! 😆

* `ensure_current_sync_id` returns an `ApiResult` in case the bridged engine persists the sync ID in SQLite or another store that can return errors.
* It returns the sync ID to use (instead of `()`), because the engine can decide it doesn't want to adopt the `sync_id` it got from the server, and wants the server to use the device's sync ID instead. I think the only engine to take advantage of that is bookmarks, on the first sync after restoring from a backup.

In just about every case, the device handles sync ID mismatches by going "my sync ID is X but you're saying it should be Y, let me take X, reset all my local sync state, and merge all your records with mine."

On a first sync after a bookmarks restore, though, the device goes "my sync ID is X, and even though you're saying it should be Y, I want you to forget Y, write X to your `meta/global`, erase everything in your `bookmarks` collection, and take all my new records."

Reply:
Thanks! I meant to delete that comment as I later realized you had already documented by the result value exists in bridged_engine.rs :)